src/plugins/ui_feature/ui_sheet.ts

```diff
@@ -74,28 +74,39 @@ export class SheetUIPlugin extends UIPlugin {
   // Getters
   // ---------------------------------------------------------------------------
 
-  getCellWidth(position: CellPosition): number {
-    const text = this.getCellText(position);
+  getCellWidth(position: CellPosition): number | undefined {
     const style = this.getters.getCellComputedStyle(position);
-    const multiLineText = text.split(NEWLINE);
-    let contentWidth = Math.max(...multiLineText.map((line) => this.getTextWidth(line, style)));
+
+    let contentWidth = 0;
+
+    const content = this.getters.getEvaluatedCell(position).formattedValue;
+    if (content) {
+      const multiLineText = getMultiLineText(this.ctx, content, style, undefined);
```

Comment:
when `width` is `undefined`, it's equivalent to just splitting the lines. Why not have a simple `splitTextLines` function ? (that `getMultiLineText` could use)

Reply:
I can do it if you really want, but that means a ternary `width ? getMultiLineText(...) : splitTextLines(...)` at two places, which I find a bit annoying.